### ISSUE TITLE
Distillation Trainer requires labels in the data

### ISSUE THREAD

**Issue by harindercnvrg** [huggingface/setfit#320]
I tried to run Distillation trainer setup as shown below:

```
from setfit import SetFitModel, SetFitTrainer, DistillationSetFitTrainer

train_dataset_student = agnews["train"].shuffle(seed=0).select(range(1000))

#Distill
student_model = SetFitModel.from_pretrained("sentence-transformers/paraphrase-MiniLM-L3-v2")
teacher_model = teacher_trainer.model
student_trainer = DistillationSetFitTrainer(teacher_model=teacher_model, train_dataset=train_dataset_student, student_model=student_model, eval_dataset=test_dataset, loss_class=CosineSimilarityLoss, metric="accuracy", batch_size=16, num_iterations=20, num_epochs=1)
```

I am getting this error

```
ValueError: A column mapping must be provided when the dataset does not contain the following columns: {'text', 'label'}
```

My dataset has only one column called "text"
```
Dataset({
    features: ['text'],
    num_rows: 1000
})
```

The whole point of distillation is not having labelled data. I think the distillation trainer does training for the distilled student body and teacher logits at the end and thus requires labels for that, but how can we provide those few labels separately?

**Comment by tomaarsen** [comment]
Hello @harindercnvrg 

You're very right, this is not desirable behaviour. When inspecting the code, I notice that it runs the same `_validate_column_mapping` that the "regular" Trainer uses, i.e. the method that requires a `text` and `labels`:
https://github.com/huggingface/setfit/blob/0420165a14431809a63ffc36f2342b6ed8190d0c/src/setfit/trainer_distillation.py#L147-L154

However, it doesn't even end up using `y_train`, it simply overrides it a little further down:
https://github.com/huggingface/setfit/blob/0420165a14431809a63ffc36f2342b6ed8190d0c/src/setfit/trainer_distillation.py#L196

With other words, as of right now a `label` column is required, but it is not used. I would apply the following:
```python
train_dataset_student = train_dataset_student.add_column("label", [0] * len(train_dataset_student))
```
Note: it doesn't have to be `0`.
I haven't been able to verify this in practice, but it should work, in theory.

- Tom Aarsen

**Comment by tomaarsen** [comment]
This should be solved once #439 merges into `main`.

**Comment by tomaarsen** [comment]
Closed via #439
